### DISCUSSION TITLE
Callback plugins in ansible-navigator

### DISCUSSION THREAD

**Discussion by fanks4** [ansible/ansible-navigator#1552] [Q&A]
When running ansible-playbooks, there is a possibility to format the output in a nice and good readable yaml format using callback plugins.
But it seems stdout_callback is ignored, wen running the ansible-navigator run subcommand instead of ansible-playbook.

You can replicate this behavior using a ansible.cfg like this:
```
[defaults]
inventory = inventory
stdout_callback = yaml
```

Running any playbook like this results in a nice output as in a yaml file:
``` 
ansible-playbook playbook.yml -K --check -v
``` 

Running any playbook with the subcommand results in a densed default output.
```
ansible-navigator run -m stdout playbook.yml -K --check -v
``` 

In my opinion a nice formating would also be helpful, when running ansible-navigator in mode stdout. What do you think?
Please let me know if i went wrong.

Best regards
fanks4

**Comment by ttyS4** [comment]
I am also bitten by this.

**Comment by chomatz** [comment]
as a workaround, i just set an environment variable in my ansible-navigator.yml
[NOTE: this works for ansible-core >= 2.13](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/default_callback.html#parameter-result_format)

```
ansible-navigator:
  execution-environment:
    environment-variables:
      set:
        ANSIBLE_CALLBACK_RESULT_FORMAT: yaml
```
